### PR TITLE
[BATCH] support ov::enable_profiling

### PR DESCRIPTION
**Pull request by himanshugupta11002**
### Details:
To complete task-I added PERF_COUNT to the plugin.cpp
And To complete the task-2 I have added get_property_params() 

# Tickets:
#23075, CVS-130236

### CODE REVIEW

**Comment by himanshugupta11002** [comment]
Hello @zhaixuejun1993 @rkazants I am eager to join GSOC-2024 with OpenVINO and plan to submit a proposal for one of the project ideas. I have already begun exploring the codebase of OpenVINO, and I'm actively working on addressing other issues. I would appreciate guidance on any additional steps I should take.

Additionally, I'm curious if you could review my pull request.

File: src/plugins/auto_batch/src/plugin.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 namespace ov {
 namespace autobatch_plugin {
 
-std::vector<std::string> supported_configKeys = {ov::device::priorities.name(), ov::auto_batch_timeout.name()};
+std::vector<std::string> supported_configKeys = {ov::device::priorities.name(), ov::auto_batch_timeout.name(), "PERF_COUNT", "NUM_STREAMS"};
```
Comment (peterchen-intel) [inline src/plugins/auto_batch/src/plugin.cpp:22]: Change to API 2.0 property
NUM_STREAMS -> ov::num_streams.name()
PERF_COUNT -> ov::enable_profiling.name()

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 2]
```diff
@@ -63,6 +63,8 @@ const std::vector<get_property_params> get_property_params_test = {
     get_property_params{ov::device::priorities.name(), true},
     get_property_params{ov::cache_dir.name(), true},
     get_property_params{ov::hint::performance_mode.name(), true},
+    {"PERF_COUNT", false},
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:66]:
```suggestion
    get_property_params{ov::enable_profiling.name(), false},                   
```

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 3]
```diff
@@ -63,6 +63,8 @@ const std::vector<get_property_params> get_property_params_test = {
     get_property_params{ov::device::priorities.name(), true},
     get_property_params{ov::cache_dir.name(), true},
     get_property_params{ov::hint::performance_mode.name(), true},
+    {"PERF_COUNT", false},                   
+    {"NUM_STREAMS", true},
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:67]: The same

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 4]
```diff
@@ -63,6 +63,9 @@ const std::vector<get_property_params> get_property_params_test = {
     get_property_params{ov::device::priorities.name(), true},
     get_property_params{ov::cache_dir.name(), true},
     get_property_params{ov::hint::performance_mode.name(), true},
+    get_property_params{ov::enable_profiling.name(), false},
+    get_property_params{ov::num_streams.name(),true}
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:67]:
```suggestion
    get_property_params{ov::num_streams.name(),false}
```

Comment (zhaixuejun1993) [reply src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:67]: You need mock the function "get_property" in setup, "get_property_params{ov::num_streams.name(),false}" no throw exception

**Comment by zhaixuejun1993** [comment]
@himanshugupta11002 thanks for this. Some thing need to do more.
1) num_stream no need.
2) pls also add set property test for perf_count, such as:
[ADD] set_property_params{{{ov::enable_profiling(true)}}, false}, in https://github.com/openvinotoolkit/openvino/blob/f5259bd61499a35f9af80287ecf78a2c023f429a/src/plugins/auto_batch/tests/unit/plugin_set_property_test.cpp#L63
3) to make the get property for pref_count pass (which didn't set by default), u need mock the get method, such as:
[ADD method]
ON_CALL(*m_plugin, get_property(StrEq("PERF_COUNT"), _))
            .WillByDefault(Return(true));
at here https://github.com/openvinotoolkit/openvino/blob/f5259bd61499a35f9af80287ecf78a2c023f429a/src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp#L40

**Review by zhaixuejun1993 — APPROVED** [review]

**Comment by zhaixuejun1993** [comment]
Pls fix the format issue @himanshugupta11002

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 5]
```diff
@@ -37,7 +37,7 @@ class GetPropertyTest : public ::testing::TestWithParam<get_property_params> {
         std::tie(m_property_name, m_throw_exception) = this->GetParam();
         m_plugin =
             std::shared_ptr<NiceMock<MockAutoBatchInferencePlugin>>(new NiceMock<MockAutoBatchInferencePlugin>());
-
+        ON_CALL(*m_plugin, get_property(StrEq("PERF_COUNT"), _)).WillByDefault(Return(true));
         ON_CALL(*m_plugin, get_property).WillByDefault([this](const std::string& name, const ov::AnyMap& arguments) {
             return m_plugin->Plugin::get_property(name, arguments);
         });
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:43]:
```suggestion
        });
        ON_CALL(*m_plugin, get_property(StrEq("PERF_COUNT"), _))
            .WillByDefault(Return(true));
```

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 6]
```diff
@@ -37,7 +37,7 @@ class GetPropertyTest : public ::testing::TestWithParam<get_property_params> {
         std::tie(m_property_name, m_throw_exception) = this->GetParam();
         m_plugin =
             std::shared_ptr<NiceMock<MockAutoBatchInferencePlugin>>(new NiceMock<MockAutoBatchInferencePlugin>());
-
+        ON_CALL(*m_plugin, get_property(StrEq("PERF_COUNT"), _)).WillByDefault(Return(true));
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:40]:
```suggestion

```

File: src/plugins/auto_batch/src/plugin.cpp [hunk 7]
```diff
@@ -19,7 +19,7 @@
 namespace ov {
 namespace autobatch_plugin {
 
-std::vector<std::string> supported_configKeys = {ov::device::priorities.name(), ov::auto_batch_timeout.name()};
+std::vector<std::string> supported_configKeys = {ov::device::priorities.name(), ov::auto_batch_timeout.name(), ov::enable_profiling.name()};
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/src/plugin.cpp:22]: pls fix this format issue

**Review by riverlijunjie — APPROVED** [review]

**Comment by riverlijunjie** [comment]
@himanshugupta11002  could you please not force push the commit to avoid overwriting the original commits and comments? The latest commit has introduced more issues.

**Comment by riverlijunjie** [comment]
> > @himanshugupta11002 could you please not force push the commit to avoid overwriting the original commits and comments? The latest commit has introduced more issues.
> 
> I accidentally discarded the updates in my master branch that closed this PR. Therefore, I reopened this PR with the previous changes. As this latest commit has more issues can you guide me how to solve them.

Maybe you can rebase to latest master branch and clean up unnecessary code, because I saw some old code garbage in your commit, such as `2023 `and `dimension_tracker` which have been removed in master branch.

File: src/plugins/auto_batch/src/plugin.cpp [hunk 8]
```diff
@@ -168,15 +170,15 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
             if (shape.is_dynamic())
                 OPENVINO_THROW("Auto-batching does not support dynamic networks!");
             // check the batch dim: either 0th (and the original batch size of 1) or none
-            if (shape.size() && shape[0].has_symbol()) {
+            if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/src/plugin.cpp:173]: Why use `ov::DimensionTracker::get_label()` to replace `shape[0].has_symbol()`

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
build_jenkins

**Comment by wangleis** [comment]
build_jenkins

File: src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp [hunk 9]
```diff
@@ -63,6 +64,8 @@ const std::vector<get_property_params> get_property_params_test = {
     get_property_params{ov::device::priorities.name(), true},
     get_property_params{ov::cache_dir.name(), true},
     get_property_params{ov::hint::performance_mode.name(), true},
+    get_property_params{ov::enable_profiling.name(), false},
+
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/tests/unit/plugin_get_property_test.cpp:68]: Please remove blank line to solve CI issue:
image

**Comment by riverlijunjie** [comment]
build_jenkins

**Review by wangleis — APPROVED** [review]

**Pull request merged**